### ISSUE TITLE
Detect `#` inline comments that used as a part of the value

### ISSUE THREAD

**Issue by Khabib73** [wemake-services/dotenv-linter#978]
# Bug report

## What's wrong

It doesn't seem right, but it's okay.:
```
KEY=VALUE #                  COMMENT
```

## How is that should be

Rule [`SpacedCommentViolation`](https://dotenv-linter.readthedocs.io/en/latest/pages/violations/index.html#dotenv_linter.violations.comments.SpacedCommentViolation) say:

> Restricts to write comment with leading or trailing spaces.

I think it's not causing an error because we are checking raw_text (not substr with a comment) ?

```python
def _check_comment_spaces(self, node: Comment) -> None:
    if node.raw_text.startswith(' ') or node.raw_text.endswith(' '):
        self._add_violation(
            SpacedCommentViolation(node, text=node.raw_text),
        )
```

### pip information

Contents of `pip freeze`:

- Python 3.14.4
- dotenv-linter 0.7.0

**Comment by sobolevn** [comment]
Yes, this should show an error: but not the one you assumed :)
This is actually not a comment, but a part of the value in `.env` files.
But, we can detect patterns like `KEY=VALUE # comment` and raise a different error.

**Comment by Khabib73** [comment]
@sobolevn  Thanks!

> But, we can detect patterns like KEY=VALUE # comment and raise a different error.

Please, put me assignees of this issue

**Comment by sobolevn** [comment]
Thanks a lot for your help!

**Comment by Khabib73** [comment]
@sobolevn 

Actually, It's not that easy. It's difficult to determine the true type of the comment. 
For example, consider `API_KEY=VALUE # MY API KEY `versus `API_KEY=9430943#A32A`. 

Since comments can be written without a space (e.g. `9430943#A32A`), it becomes unclear how to distinguish a real comment from a sequence of characters.

**Comment by sobolevn** [comment]
Let's detect ones that have spaces at least: `Value # comment`

**Comment by sobolevn** [comment]
We can also add `# dotenv:ignore[ERROR]` comments on top of the values (the line above) that can be false positives :)
This can be a separate feature.

**Comment by Khabib73** [comment]
> We can also add # dotenv:ignore[ERROR] comments on top of the values (the line above) that can be false positives :)
This can be a separate feature.

Can we create a issue?

**Comment by sobolevn** [comment]
Sure!
